src/test/java/org/example/unit/JobRoleServiceTest.java

```diff
@@ -0,0 +1,158 @@
+
+    Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+    Mockito.when(jobRoleDao.getAllRoles(conn)).thenReturn(expectedResult);
+
+    List<JobRole> result = jobRoleService.getAllRoles();
+
+    assertEquals(expectedResult.get(0).getJobRoleId(),
+            result.get(0).getJobRoleId());
+    assertEquals(expectedResult.get(0).getCapability(),
+            result.get(0).getCapability());
+    assertEquals(expectedResult.get(0).getJobFamily(),
+            result.get(0).getJobFamily());
+    assertEquals(expectedResult.get(0).getJobProfileTitle(),
+            result.get(0).getJobProfileTitle());
+    assertEquals(expectedResult.get(0).getManagementLevel(),
+            result.get(0).getManagementLevel());
+    assertEquals(expectedResult.get(0).getDescription(),
+            result.get(0).getDescription());
+    assertEquals(expectedResult.get(0).getMinimalEssentialRequirements(),
+            result.get(0).getMinimalEssentialRequirements());
+    assertEquals(expectedResult.get(1).getJobRoleId(),
+            result.get(1).getJobRoleId());
+    assertEquals(expectedResult.get(1).getCapability(),
+            result.get(1).getCapability());
+    assertEquals(expectedResult.get(1).getJobFamily(),
+            result.get(1).getJobFamily());
+    assertEquals(expectedResult.get(1).getJobProfileTitle(),
+            result.get(1).getJobProfileTitle());
+    assertEquals(expectedResult.get(1).getManagementLevel(),
+            result.get(1).getManagementLevel());
+    assertEquals(expectedResult.get(1).getDescription(),
+            result.get(1).getDescription());
+    assertEquals(expectedResult.get(1).getMinimalEssentialRequirements(),
+            result.get(1).getMinimalEssentialRequirements());
+  }
+
+  /*
+  Unit test for the getSpecificationById method
+
+  When the dao throws an SpecificationDoesNotExist Exception
+
+  Expect SpecificationDoesNotExist to be thrown
+      */
+  @Test
+  void getSpecificationById_shouldThrowSpecificationDoesNotExistException_whenDaoReturnsNull()
+          throws SQLException, DatabaseConnectionException {
+    Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+    int id = 1000000;
+    Mockito.when(jobRoleDao.getSpecificationById(id, conn)).thenReturn(null);
+    assertThrows(SpecificationDoesNotExistException.class,
+            () -> jobRoleService.getSpecificationById(id));
+  }
+
+  @Test
+  void getSpecificationById_shouldReturnId_whenDaoReturnsId() throws DatabaseConnectionException,
+          SQLException, SpecificationDoesNotExistException, FailedToGetSpecificationException {
+    int roleId = 1;
+    JobRole specification = new JobRole(roleId, "test", "test", "test", "test", "test", "test");
+    Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+    Mockito.when(jobRoleDao.getSpecificationById(roleId, conn)).thenReturn(specification);
+    assertEquals(specification, jobRoleService.getSpecificationById(roleId));
+  }
+}
```

Comment:
Needs an extra line at the bottom, could you also add a test for `SQLException` or `DatabaseConnectionException`